common/src/test/java/com/revenuecat/purchases/common/BackendTest.kt

```diff
@@ -1451,6 +1492,202 @@ class BackendTest {
         assertThat(headersSlot.captured["marketplace"]).isEqualTo("DE")
     }
 
+    // region postDiagnostics
+
+    @Test
+    fun `postDiagnostics makes call with correct parameters`() {
+        val diagnosticsList = listOf(JSONObject("{\"test-key\":\"test-value\"}"))
+        backend.postDiagnostics(diagnosticsList, {}, { _, _ -> })
+        verify(exactly = 1) {
+            mockClient.performRequest(
+                baseURL = mockDiagnosticsBaseURL,
+                path = "/telemetry",
+                body = mapOf("entries" to JSONArray(diagnosticsList)),
+                requestHeaders = mapOf("Authorization" to "Bearer TEST_API_KEY")
+            )
+        }
+    }
+
+    @Test
+    fun `postDiagnostics only executes once same request if one in progress`() {
+        val diagnosticsList = listOf(JSONObject("{\"test-key\":\"test-value\"}"))
+        mockResponse(
+            path = "/telemetry",
+            body = null,
+            responseCode = 200,
+            clientException = null,
+            resultBody = null,
+            delayed = true,
+            baseURL = mockDiagnosticsBaseURL
+        )
+        val lock = CountDownLatch(3)
+        asyncBackend.postDiagnostics(diagnosticsList, { lock.countDown() }, { _, _ -> fail("expected success") })
+        asyncBackend.postDiagnostics(diagnosticsList, { lock.countDown() }, { _, _ -> fail("expected success") })
+        asyncBackend.postDiagnostics(diagnosticsList, { lock.countDown() }, { _, _ -> fail("expected success") })
+        lock.await(2000, TimeUnit.MILLISECONDS)
+        assertThat(lock.count).isEqualTo(0)
+        verify(exactly = 1) {
+            mockClient.performRequest(
+                baseURL = mockDiagnosticsBaseURL,
+                path = "/telemetry",
+                body = mapOf("entries" to JSONArray(diagnosticsList)),
+                requestHeaders = mapOf("Authorization" to "Bearer TEST_API_KEY")
+            )
+        }
+    }
+
+    @Test
+    fun `postDiagnostics executes same request if done after first one finishes`() {
```

Comment:
I imagine that this is to play it safe, but in practice we should never have more than one postDiagnostics request happening in the same session, let alone at the same time

Reply:
You're completely right. But yeah, this is just in case.